### PR TITLE
Various Fixes

### PR DESCRIPTION
**Pull request by orangeturtle739**
* Hashes
* Deterministic Verilog
* sext/zext slicing fixes

### CODE REVIEW

File: pymtl/tools/translation/verilator_wrapper.templ.py [hunk 1]
```diff
@@ -91,7 +94,7 @@ def elaborate_logic( s ):
     verilator_vcd_file = ""
     if s.vcd_file:
       filen, ext         = os.path.splitext( s.vcd_file )
-      verilator_vcd_file = '{{}}.verilator1{{}}'.format(filen, ext)
+      verilator_vcd_file = '{{}}.verilator{{}}'.format(filen, ext)
```
Comment (orangeturtle739) [inline pymtl/tools/translation/verilator_wrapper.templ.py]: @cbatten Why did you add a 1 here https://github.com/cornell-brg/pymtl/commit/15926e9c87f1fc76a1335e63930a138db3ca71aa?

File: pymtl/tools/translation/verilog_behavioral.py [hunk 2]
```diff
@@ -541,7 +541,7 @@ def visit_Call(self, node):
       sig_name   = self.visit( node.args[0] )
       sig_nbits  = node.args[0]._object.nbits
       ext_nbits  = self.visit( node.args[1] )
-      return '{{ {{ {3}-{1} {{ {0}[{2}] }} }}, {0}[{2}:0] }}' \
+      return '{{ {{ {3}-{1} {{ {0}[{2}] }} }}, {0} }}' \
```
Comment (orangeturtle739) [inline pymtl/tools/translation/verilog_behavioral.py]: This extra slice here is unnecessary as far as I can tell, and causes it to generate verilog that, while accepted by verilator, is not accepted by vivado.

File: pymtl/tools/simulation/vcd.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def mangle_name( name ):
 def write_vcd_signal_defs( o, model ):
 
   vcd_symbol = _gen_vcd_symbol()
-  all_nets   = set()
+  all_nets   = []
```
Comment (orangeturtle739) [inline pymtl/tools/simulation/vcd.py]: I fixed this in https://github.com/cornell-brg/pymtl/pull/180, someone changed it back...

**Comment by cbatten** [comment]
I am going to go ahead and merge and let's see if it causes issues with anyone. Thanks for your hard work on this!
